### PR TITLE
Fix/operator agentversion never updated

### PR DESCRIPTION
**Pull request by ycliuhw**
## Description of change

Fix: operators' `upgradedToVersion` in `agent.cfg` is not updated after `uprade-model`;

## QA steps

- prepare caas model;
- deploy workloads;
- upgrade-controller;
- upgrade-model;
- check `upgradedToVersion`;

```bash
$ mkubectl exec gitlab-k8s-operator-0  -n t1 -it cat /var/lib/juju/agents/application-gitlab-k8s/agent.conf | grep upgradedToVersion
```

## Documentation changes

None;

## Bug reference

No, not released yet;

### CODE REVIEW

**Comment by ycliuhw** [comment]
[operator agent.conf targetAgentVersion has to be updated after model upgrade](https://trello.com/c/VEIdcqvI/709-operator-agentconf-targetagentversion-has-to-be-updated-after-model-upgrade)

File: worker/upgradesteps/manifold.go [hunk 1]
```diff
@@ -56,32 +58,39 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
...
-			if err != nil {
-				return nil, err
-			}
-			entity, err := agentFacade.Entity(agent.CurrentConfig().Tag())
-			if err != nil {
-				return nil, err
+			var jobs []multiwatcher.MachineJob
+			if names.IsValidApplication(agent.CurrentConfig().Tag().String()) {
+				// application tag for CAAS operator.
+				jobs = []multiwatcher.MachineJob{}
```
Comment (wallyworld) [inline worker/upgradesteps/manifold.go]: We don't need an empty slice do we?
Can't we just leave it as nil? If not let's fix it so we can and avoid the multiwatcher import.

File: worker/upgradesteps/manifold.go [hunk 2]
```diff
@@ -56,32 +58,39 @@ func Manifold(config ManifoldConfig) dependency.Manifold {
...
 			// TODO(fwereade): use appropriate facade!
-			agentFacade, err := apiagent.NewState(apiConn)
-			if err != nil {
-				return nil, err
-			}
-			entity, err := agentFacade.Entity(agent.CurrentConfig().Tag())
-			if err != nil {
-				return nil, err
+			var jobs []multiwatcher.MachineJob
+			if names.IsValidApplication(agent.CurrentConfig().Tag().String()) {
```
Comment (wallyworld) [inline worker/upgradesteps/manifold.go]: You can just check Tag().Kind(), less verbose. Here and below

File: worker/upgradesteps/worker.go [hunk 3]
```diff
@@ -209,6 +209,9 @@ func (w *upgradesteps) run() error {
 			return errors.Trace(err)
 		}
 		w.isCaas = model.Type() == state.ModelTypeCAAS
+	} else {
+		// application tag for CAAS operator.
+		w.isCaas = names.IsValidApplication(w.agent.CurrentConfig().Tag().String())
```
Comment (wallyworld) [inline worker/upgradesteps/worker.go]: If we are doing htis can we just have one check for isCass that can be used for controller or not.
Having 2 checks is a bit clunky.

File: worker/upgradesteps/manifold.go (hunk 1 above)
Comment (ycliuhw) [reply worker/upgradesteps/manifold.go]: sure, fixed,

File: worker/upgradesteps/manifold.go (hunk 2 above)
Comment (ycliuhw) [reply worker/upgradesteps/manifold.go]: yup, now it's using Kind to check

File: worker/upgradesteps/worker.go (hunk 3 above)
Comment (ycliuhw) [reply worker/upgradesteps/worker.go]: yup, moved this to manifold then pass in

**Comment by ycliuhw** [comment]
$$merge$$

**Pull request merged**